subprojects/base-services/src/main/java/org/gradle/internal/service/DefaultServiceRegistry.java

```diff
@@ -255,61 +252,27 @@ public DefaultServiceRegistry addProvider(Object provider) {
      * Closes all services for this registry. For each service, if the service has a public void close() or stop() method, that method is called to close the service.
      */
     public void close() {
-        if (state == State.CLOSED || state == State.CLOSING) {
-            return;
-        }
-
-        synchronized (stateLock) {
-            if (state == State.CLOSED || state == State.CLOSING) {
-                return;
-            }
-            state = State.CLOSING;
-            waitForPendingRequests();
+        noLongerMutable();
+        if (state.compareAndSet(State.IN_USE, State.CLOSED)) {
```

Comment:
Just as a general comment, I find `IN_USE` to be somewhat confusing as it implies to me that something is actively using the registry.  Something like `READY` or `STARTED` might be more clear.  I don't think this is important enough to require a change, unless you feel compelled to do so, it's just a comment.

Reply:
Good point, I'll change that.